app/services/api/webflow-helpers.ts

```diff
@@ -1,141 +1,193 @@
     },
     {
       date: "2023-05-19",
       title: "Conference Day 2",
-      events: convertScheduleToScheduleCard(scheduleData, "2023-05-19"),
+      events: convertScheduleToScheduleCard(events, "Friday"),
     },
   ]
 }
 
-export const convertWorkshopToScheduleCard = (
-  workshopData?: WorkshopProps[],
-): ScheduleCardProps[] => {
-  return workshopData?.map((workshop, index) => {
-    const sources = [workshop["instructor-info"]["speaker-photo"].url]
-    if (workshop["second-instructor-2"]) {
-      sources.push(workshop["second-instructor-2"]["speaker-photo"].url)
-    }
-    return {
-      variant: "workshop",
-      // Temporary solution for workshop time for testing centering the card on the screen
-      time: `${index}:00`,
-      eventTitle: "workshop",
-      heading: workshop.name,
-      subheading: workshop["instructor-info"].name,
-      sources,
-      level: workshop.level,
-      id: workshop._id,
-    }
-  })
+const convertScheduleToCardProps = (schedule: ScheduledEvent): ScheduleCardProps => {
+  switch (schedule.type) {
+    case "Recurring":
+      return {
+        variant: "recurring",
+        time: formatDate(schedule["day-time"], "h:mm a"),
+        eventTitle: schedule["recurring-event"]?.name,
+        heading: "",
+        subheading: schedule["recurring-event"]?.["event-description"],
+        sources: [],
+        id: schedule._id,
+      }
+    case "Party":
+      return {
+        variant: "party",
+        time: formatDate(schedule["day-time"], "h:mm a"),
+        eventTitle: "party",
+        heading: schedule.name,
+        subheading: schedule["break-party-description"],
+        sources: [],
+        id: schedule._id,
+      }
+    case "Talk":
+      return {
+        variant: "talk",
+        time: formatDate(schedule["day-time"], "h:mm a"),
+        eventTitle: schedule.type,
+        heading: schedule.talk?.name,
+        subheading: schedule.talk?.description,
+        sources: schedule.talk?.["speaker-s"]?.map((s) => s["speaker-photo"].url) ?? [],
+        id: schedule._id,
+      }
+    case "Speaker Panel":
+      return {
+        variant: "talk",
+        time: formatDate(schedule["day-time"], "h:mm a"),
+        eventTitle: schedule.type,
+        heading: schedule.talk?.["speaker-s"]?.map((s) => s.name).join(", ") ?? "",
+        subheading: "",
+        sources: schedule.talk?.["speaker-s"]?.map((s) => s["speaker-photo"].url) ?? [],
+        id: schedule._id,
+      }
+    case "Workshop":
+      // eslint-disable-next-line no-case-declarations
+      const workshop = schedule.workshop
+      return {
+        variant: "workshop",
+        time: formatDate(schedule["day-time"], "h:mm a"),
+        eventTitle: "workshop",
+        heading: workshop?.name,
+        subheading: workshop?.["instructor-info"]?.name,
+        sources: [
+          workshop?.["instructor-info"]?.["speaker-photo"].url,
+          // ...(workshop?.assistants?.map((a) => a["speaker-photo"].url) ?? []),
```

Comment:
I think the plan is not to show multiple speakers on the workshop card and only show everyone the details so it will probably make sense to remove this line :)